doc/source/tutorial/stats/rv_infrastructure.md

```diff
@@ -0,0 +1,920 @@
+
+First, note that MLE is one of several potential strategies for fitting distributions to data. It does nothing more than find the values of the distribution parameters for which the negative log-likelihood (NLLF) is minimized. Note that for the original data, the NLLF is:
+
+```{code-cell} ipython3
+loc = 0
+stats.weibull_min.nnlf((c, loc, scale), data)
+```
+
+The NLLF of the parameters estimated using `fit` are lower:
+
+```{code-cell} ipython3
+stats.weibull_min.nnlf(params, data)
+```
+
+Therefore, `fit` considers its job done, whether or not this satisfies the user's notions of a "good fit" or whether the parameters are within reasonable bounds. It gives the user no control over the process (other than specifying a guess).
+
++++
+
+Another problem is that MLE is an inherently problematic for some distributions. In this example, for instance, we can make the NLLF as low as we please simply by setting the location to the minimum of the data - even for bogus values of the shape and scale parameters.
+
+```{code-cell} ipython3
+stats.weibull_min.nnlf((0.1, np.min(data), 10), data)
+```
+
+Compounding with these issues is that `fit` estimates all distribution parameters by default. In the example above, we are probably only interested in the more common two-parameter Weibull distribution because the actual location is zero. `fit` *can* accommodate this by specifing that the location is fixed parameter.
+
+```{code-cell} ipython3
+# params = stats.weibull_min.fit(data, loc=0)  # careful! this provides loc=0 as a *guess*
+params = stats.weibull_min.fit(data, floc=0)
+params
+```
+
+```{code-cell} ipython3
+stats.weibull_min.nnlf(params, data)
+```
+
+By relying on a `fit` method to "just work", users are encouraged to neglect some of these important details. 
+
+Instead, we suggest that it is almost as simple to fit distributions to data using the generic optimization tools provided by SciPy. However, this allows the user to perform the fit according to their preferred objective function, provides options when things don't work as expected, and helps the user to avoid the most common pitfalls.
+
+For instance, suppose our desired objective function is indeed the negative log likelihood function. It is simple to define as a function of the shape parameter `c` and scale only, leaving the location at its default of zero.
+
+```{code-cell} ipython3
+def nllf(params):
+    c, scale = params
+    X = Weibull(c=c) * scale
+    return -X.logpdf(data).sum()
+
+c_, loc_, scale_ = params
+nllf((c_, scale_))
+```
```

Comment:
Using params both as a 2-tuple and 3-tuple in the same codeblock is confusing IMO. I would keep the custom `nllf` as using a 3-typle, and then just discard `loc` when unpacking `params`:
```
    c, _, scale = params
```

Reply:
I will try to make that more clear. It will need to be just 2-tuple in the optimization, though.

Reply:
I would still prefer that the `params` in the signature of the (here only local) `nllf` matches the 3-tuple params in the rest of the notebook, rather than introducing a 2-tuple variant just for this code-block

Reply:
To make sure we're talking about the same thing, it's not just in this code block: there are only two parameters being optimized, so all the code blocks that define an object function do this. 

So we would have to show `loc` being added as a decision variable and then constrained to be equal to 0. Many solvers have trouble with that. We happened to have given some `minimize` solvers the ability to handle that, but not all, so I would rather avoid show that. 

Can you think of other things that would address your concern? For example, usually in objective functions, the variable is just called `x`. I can change the name of the argument of objective functions to just `x`?

Reply:
If we want to be completely unambiguous we could use `decision_variables` instead of `params`. That should make sure no one would mistakenly assume that `params` had some specific meaning other than just the variables being optimized.

Reply:
Hmm, but that is still a bit ambiguous because there is a distinction between the variables themselves and the value at that iteration. Maybe `values_of_decision_variables_at_this_iteration`?

Reply:
Actually, I had already removed three-element use of `params` before the last commit. I was just immediately unpacking calls to `weibull_min.fit` into `c_, loc_, scale_`. I went ahead and removed the remaining uses of `params`, though, for good measure.